### PR TITLE
Feature mes 2324 next batch implementation

### PR DESCRIPTION
**Pull request by MarkPatt89**
# Description and relevant Jira numbers
- Create a query to return test result json
- Added repositories to handle each interface type (only tars implemented)
- Compress data
## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [x] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: src/functions/getNextUploadBatch/framework/repositories/tars-batch-repository.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import { TestResultRecord } from '../../domain/test-results';
+
+export class TarsBatchRepository implements IBatchRepository {
+
+  getUploadQueueData(batchSize: number): Promise<TestResultRecord[]> {
+    const connection: mysql.Connection = getConnection();
+    let batch;
+    try {
+      batch = connection.promise().query(buildTarsNextBatchQuery(batchSize));
+      console.log(batchSize);
```
Comment (roberthall12) [inline src/functions/getNextUploadBatch/framework/repositories/tars-batch-repository.ts]: `console.log`

Comment (MarkPatt89) [reply src/functions/getNextUploadBatch/framework/repositories/tars-batch-repository.ts]: Done

File: src/functions/getNextUploadBatch/domain/validation-warnings.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export enum ValidationWarnings {
+  PARAM_NOT_PROVIDED = 'Invalid APIGatewayEvent params provided',
```
[the remark this answers is not in the document]
Comment (MarkPatt89) [reply src/functions/getNextUploadBatch/domain/validation-warnings.ts]: Changed.

File: src/functions/getNextUploadBatch/framework/handler.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import { gzipSync } from 'zlib';
+
+export async function handler(event: APIGatewayEvent, fnCtx: Context): Promise<Response> {
+  let nextBatchData;
+  const interfaceType = convertToInterfaceType(event.queryStringParameters.interface);
+  const batchSize = event.queryStringParameters.batch_size;
+  const validationResponse = validateQueryParameters(interfaceType, batchSize);
+
+  if (validationResponse !== undefined) {
+    return createResponse(validationResponse, HttpStatus.BAD_REQUEST);
```
Comment (ammarhaiderbjss) [inline src/functions/getNextUploadBatch/framework/handler.ts]: Could bring in a simple joi schema here to validate types on the incoming event, then we could return a useful error message to the caller.

Comment (MarkPatt89) [reply src/functions/getNextUploadBatch/framework/handler.ts]: Yes, that actually does sound good. Let me have a play around with it.

**Pull request merged**